src/NServiceBus.Core/Pipeline/Incoming/InvokeHandlerContext.cs

```diff
@@ -51,6 +52,11 @@ public InvokeHandlerContext(MessageHandler handler, string messageId, string rep
         public SynchronizedStorageSession SynchronizedStorageSession => Get<SynchronizedStorageSession>();
 
         /// <summary>
+        /// Token
+        /// </summary>
+        public CancellationToken CancellationToken => this.Get<CancellationTokenSource>().Token;
```

Comment:
I like it.

On Thu, Dec 17, 2015 at 5:02 AM, John Simons <notifications@github.com>
wrote:

> In src/NServiceBus.Core/Pipeline/Incoming/InvokeHandlerContext.cs
> <https://github.com/Particular/NServiceBus/pull/3242#discussion_r47867148>
> :
>
> > @@ -51,6 +52,11 @@ public InvokeHandlerContext(MessageHandler handler, string messageId, string rep
> >          public SynchronizedStorageSession SynchronizedStorageSession => Get<SynchronizedStorageSession>();
> >
> >          /// <summary>
> > +        /// Token
> > +        /// </summary>
> > +        public CancellationToken CancellationToken => this.Get<CancellationTokenSource>().Token;
>
> Should we cache the token ?, so in other words call
> this.Get<CancellationTokenSource>().Token only once.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3242/files#r47867148>.
>



-- 
Mauro Servienti | Solution Architect | Particular Software